src/server/game/Entities/Object/Object.cpp

```diff
@@ -1486,8 +1486,7 @@ void WorldObject::UpdateAllowedPositionZ(float x, float y, float& z, float* grou
     {
         if (!unit->CanFly())
         {
-            Creature const* c = ToCreature();
-            bool canSwim = c ? c->CanSwim() : true;
+            bool canSwim = unit->CanSwim();
```

Comment:
can be `auto` here

Reply:
I prefer to explicitly define it, I don't see any advantages in doing auto with booleans